core/primitives/src/crypto/signer.rs

```diff
@@ -198,36 +204,62 @@ impl InMemorySigner {
         let mut rng = OsRng::new().expect("Unable to generate random numbers");
         let account_id: String =
             rng.sample_iter(&Alphanumeric).filter(|c| c.is_ascii_alphabetic()).take(10).collect();
-        let (public_key, secret_key) = signature::get_key_pair();
+        let (public_key, secret_key) = get_key_pair();
         let bls_secret_key = BlsSecretKey::generate();
         let bls_public_key = bls_secret_key.get_public_key();
         Self { account_id, public_key, secret_key, bls_public_key, bls_secret_key }
     }
 }
 
+// Use hash for the Key so that we do not store raw crypto keys in the memory. Also more memory
+// efficient.
+cached_key!{
+  TRANSACTION_CACHE: SizedCache<CryptoHash, Signature> = SizedCache::with_size(1_000_000);
```

Comment:
This is for the wallet of the user running the node?
What's the point of caching?

Reply:
E.g. for unit tests.

Reply:
for tests that sign the same transaction many times?

Reply:
Yes, now that you point it, I do not think it makes sense here.